src/MessagingThreadListItem/MessagingThreadListItem.tsx

```diff
@@ -26,6 +26,11 @@ const cssClasses = {
   UNREAD_TEXT: "ThreadListItem--UnreadText",
 };
 
+const unreadOrbColorClasses = {
+  primary_blue: "ThreadListItem--UnreadOrb--PrimaryBlue",
+  accent_teal: "ThreadListItem--UnreadOrb--AccentTeal",
+};
```

Comment:
this feels a bit unorthodox, but worked so i thought what the heck. open to other ideas tho if there's a better way to handle this

Reply:
This seems pretty solid to me! I think this would have been the de-facto way to do it until recently when Typescript introduced this new feature called "Records." So, if you're interested i can describe it below but you could use your current approach as well - 

https://www.typescriptlang.org/docs/handbook/utility-types.html#example-3

A Record is a type that maps a set of strings to some other values, and enforces that each string in the set has to have a corresponding value.

So it would be something like:

```
type UnreadOrbColors = "accent_teal" | "primary_blue"

const unreadOrbColorClasses: Record<UnreadOrbColors, String> = {
   accent_teal: "ThreadListItem--UnreadOrb--AccentTeal",
  primary_blue: "ThreadListItem--UnreadOrb--PrimaryBlue",
}
```

and then in the props for the component, you would just reference the same type:

```
type Props = {
   ...
   unreadOrbColor?: UnreadOrbColor;
}

and then reference it the same way you already are with:

unreadOrbColor && unreadOrbColorClasses[unreadOrbColor],

```

The nice thing about this is that if someone added in another `unreadOrbColor`, they would get an error at `unreadOrbColorClasses` if they didnt also add the class since the Record enforces that each string in the type has a corresponding value in the record.

Reply:
Nice, I like that, very TypeScripty™️. I'll do it